From 17a3d1d047d4dfef7a5d4a922c7356d9dd24b4b6 Mon Sep 17 00:00:00 2001
From: Otto Modinos <ottomodinos@gmail.com>
Date: Tue, 28 May 2024 13:21:19 +0300
Subject: [PATCH 1/2] fix for system `db` lacking a `.pc` file

In Void Linux, we link our `kdb` against the system `db` library which does not
have a pkg-config file and the samba build scripts just cannot handle that.
---
 wscript_configure_system_mitkrb5 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/wscript_configure_system_mitkrb5 b/wscript_configure_system_mitkrb5
index d40bb36737e..9e990fe8eaf 100644
--- a/wscript_configure_system_mitkrb5
+++ b/wscript_configure_system_mitkrb5
@@ -66,6 +66,8 @@ if conf.env.KRB5_CONFIG:
         if conf.CHECK_CFG(path=conf.env.KRB5_CONFIG, args="--cflags --libs",
                           package="kdb", uselib_store="KDB5"):
             krb5_define_syslib(conf, "kdb5", conf.env['LIB_KDB5'])
+            if 'db' in conf.env['LIB_KDB5'] and not conf.check_cfg(package="db", mandatory=False):
+                krb5_define_syslib(conf, "db", [])
             conf.CHECK_HEADERS('kdb.h', lib='kdb5')
         else:
             raise Errors.WafError('Unable to find required kdb5 library!')
-- 
2.45.1

